homeassistant/helpers/trigger.py

```diff
@@ -1022,6 +1061,13 @@ async def async_get_all_descriptions(
     new_descriptions_cache = descriptions_cache.copy()
     for missing_trigger in missing_triggers:
         domain = triggers[missing_trigger]
+        if (
+            domain in _EXPERIMENTAL_TRIGGER_PLATFORMS
+            and not labs.async_is_preview_feature_enabled(
+                hass, "automation", "new_triggers_conditions"
+            )
+        ):
+            continue
```

Comment:
I think the cache will always be considered invalid if we skip a trigger here, if the general approach in the PR is OK that needs to be fixed